waddrmgr/address.go

```diff
@@ -113,6 +116,10 @@ var _ ManagedPubKeyAddress = (*managedAddress)(nil)
 // used by the caller without worrying about it being zeroed during an address
 // lock.
 func (a *managedAddress) unlock(key EncryptorDecryptor) ([]byte, error) {
+	if a.WatchingOnly() {
+		return nil, managerError(ErrWatchingOnly, errWatchingOnly, nil)
```

Comment:
Are there any locations where managed addresses are unlocked before being checked to be watching-only?  If so, returning the error up the stack would be incorrect.